<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <arg name="camera" default="/usb_cam" />

    <!-- Play rosbag record -->
    <!--include file="$(find lidar_camera_calibration)/launch/play_rosbag.launch">
        <arg name="bagfile" value="2019-09-24-13-56-36.bag" />
    </include-->

    <!-- Nodelet manager for this pipeline -->
    <node
        pkg="nodelet"
        type="nodelet"
        args="manager"
        name="lidar_camera_manager"
        output="screen" />

    <node
        pkg="image_proc"
        type="image_proc" 
        name="image_proc_node1" />

    <!-- Run image_proc/rectify nodelet -->
    <node
        pkg="nodelet"
        type="nodelet"
        name="rectify_color"
        args="load image_proc/rectify lidar_camera_manager --no-bond" >
        
        <!-- Remap input topics -->
        <remap from="image_mono" to="$(arg camera)/image_raw" />
        <remap from="camera_info" to="$(arg camera)/camera_info" />
        <!-- Remap output topics -->
        <remap from="image_rect" to="$(arg camera)/image_rect_color" />
    </node>

    <!-- Setup params for Camera-LiDAR calibration script -->
    <param name="camera_info_topic" type="str" value="$(arg camera)/camera_info" />
    <param name="image_rect_topic" type="str" value="$(arg camera)/image_rect_color" />
    <param name="velodyne_points_topic" type="str" value="/os1_cloud_node/points" />
    <param name="tf_name" type="str" value="front_fhd_cam_to_ouster_lidar_extrinsic" />
    <param name="child_frame" type="str" value="front_fhd_cam" />
    <param name="parent_frame" type="str" value="os1_lidar" />

    <!-- Run Camera-LiDAR calibration script -->
    <node   
        pkg="lidar_camera_calibration"
        type="calibrate_camera_lidar.py"
        name="calibrate_camera_lidar"
				output="screen" 
				args="--calibrate"/>

</launch>
